src/runtime.rs

```diff
@@ -183,35 +176,38 @@ impl Runtime {
             .wallets
             .get(name)
             .ok_or(RuntimeError::WalletUnknown(name.clone()))?;
-        RgbWallet::with(descr.clone(), &mut self.resolver).map_err(RuntimeError::from)
+        Ok(RgbWallet::new(descr.clone()))
     }
 
     pub fn import_contract(
         &mut self,
         contract: Contract,
+        resolver: &mut BlockchainResolver,
```

Comment:
Do we really need to take the resolver as a mutable reference? By looking to the code I cannot see where there's the need to change the struct. Making it immutable it would simplify its usage.
Same question applies to all runtime methods that take the resolver as a mutable reference

Reply:
The reason is that I assume resolver can be a caching resolver, significantly improving validation performance. I mean for the first time the resolver is asked for certain tx it goes to electrum/esplora, caches the answer - and the next time it uses the cache. Such a resolver need to be mutable.

Reply:
So right now it's not necessary but in the near future it will be, therefore it makes sense defining it as a mutable reference right away

Reply:
Yes, since otherwise it will be hard to adopt change to mutable type downstream in the future - and introduction of a simple caching resolver (which is trivial to do as a wrapper type and is not a breaking change) will become a breaking change requiring major version increase.